### PR TITLE
FIX-1591,1592 : Filter screen style changes  & Advance filter settings

### LINKED ISSUE #1591: Filter screen style changes
**Old UI**
filter-screen
**New UI**
WE-02a
- When a user is successfully connected to a server, the user view the Well and its details.
- User can click on the filter icon to
   - Limit the number of wells .
   - View the active wells, wellbores and growing logs.
   - Hide the inactive time curves.
- When the user clicks on the Show more settings, it will show the advanced filter settings where the well objects can be viewed.
- When selected or edited, the filter preference should auto saved to the users local cache.

### PR DESCRIPTION
**Pull request by harigopalmg**
## Fixes
This pull request fixes #1591 and #1592 

## Description
GUI Improvements in FIlter Panel

## Type of change
_GUI Improvements_

- Enhancement of existing functionality
- This change requires a documentation update

## Impacted Areas in Application
_Frontend:_

- Style Changes
- Functionality CHnages

## Checklist:
_Please tick all the boxes or remove the ones that aren't needed and explain why_

*Communication*
* [x] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [x] New code is covered by passing tests

## Further comments
PR for 2 issues were added together since they have cross dependency with each other

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx [hunk 1]
```diff
@@ -26,111 +46,128 @@ const FilterPanel = (): React.ReactElement => {
...
-                  dispatchNavigation({ type: NavigationType.SetCurveThreshold, payload: { curveThreshold: { ...curveThreshold, hideInactiveCurves: event.target.checked } } })
-                }
-                value={curveThreshold}
-              />
-            }
-            label={"Hide inactive time curves"}
-          />
+          <div style={{ paddingTop: "0.75rem" }}>
+            <Checkbox
+              id="filter-isActive"
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx:73]: The checkbox becomes unchecked after hiding the FilterPanel, even though isActive is set to true.

File: Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx [hunk 2]
```diff
@@ -26,111 +46,128 @@ const FilterPanel = (): React.ReactElement => {
...
+              onChange={(event) => dispatchNavigation({ type: NavigationType.SetFilter, payload: { filter: { ...filter, isActive: event.target.checked } } })}
+              value={"Show active Wells / Wellbores"}
+              color={"primary"}
+              style={{ height: "0.625rem", userSelect: "none" }}
+              label={"Show active Wells / Wellbores"}
+            />
+          </div>
+          <div style={{ userSelect: "none" }}>
+            <Checkbox
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx:83]: The checkbox becomes unchecked after hiding the FilterPanel, even though objectGrowing is set to true.

File: Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx [hunk 3]
```diff
@@ -26,111 +46,128 @@ const FilterPanel = (): React.ReactElement => {
...
+              onChange={(event: ChangeEvent<HTMLInputElement>) =>
+                dispatchNavigation({ type: NavigationType.SetCurveThreshold, payload: { curveThreshold: { ...curveThreshold, timeInMinutes: Number(event.target.value) } } })
+              }
+              value={curveThreshold.timeInMinutes}
+              autoComplete={"off"}
+            />
+          </div>
+
+          <div style={{ userSelect: "none" }}>
+            <Checkbox
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx:127]: The checkbox becomes unchecked after hiding the FilterPanel, even though hideInactiveCurves is set to true.

File: Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx [hunk 4]
```diff
@@ -26,111 +46,128 @@ const FilterPanel = (): React.ReactElement => {
     setCurveThreshold(selectedCurveThreshold);
   }, [selectedCurveThreshold]);
 
+  const wellObjectList = Object.keys(WellboreObjects).map((wellObj: string, index: number) => {
+    return (
+      <EdsProvider key={index}>
+        <Checkbox label={wellObj} style={{ height: "0.75rem" }} defaultChecked key={index} />
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx]: These checkboxes don't do anything. #1592 specifies that "Only the selected well objects are displayed in the wellbore section".

**Comment by janburak** [comment]
~~#1591 specifies that "When selected or edited, the filter preference should be auto saved to the users local cache". None of the filters seem to be saved to the cache.~~
To be determined by Thomas. Initially it is not required to cache.

File: Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx (hunk 4 above)
Comment (janburak) [reply Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx]: Checkboxes of objects that are not yet implemented should be unchecked and disabled. Well and wellbore should be checked and disabled, given that it does not make much sense to toggle them off.

File: Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx (hunk 2 above)
Comment (harigopalmg) [reply Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx:83]: storage updated

File: Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx (hunk 3 above)
Comment (harigopalmg) [reply Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx:127]: updated

File: Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx (hunk 4 above)
Comment (harigopalmg) [reply Src/WitsmlExplorer.Frontend/components/Sidebar/FilterPanel.tsx]: updated

**Comment by janburak** [comment]
Thank you for your contribution. I will take over the pull request to make some final changes and merge it.

**Review by janburak — APPROVED** [review]

**Pull request merged**